frontend/src/pages/RenderPage.js

```diff
@@ -31,6 +34,29 @@ const RenderPage = () => {
     }
   }
 
+  const navigate = useNavigate()
+
+  const leaveRoom = async () => {
+    try {
+      console.log('deleteing room with id: ' + location.state.room)
+      const res = await deleteRoomSvc(location.state.room)
+      matchingSocket.emit('leave-room', location.state.room, 'partner left')
+      console.log(JSON.stringify(res.data))
+    } catch (err) {
+      console.log(err)
+    }
+    navigate('/home')
+  }
+
+  useEffect(() => {
+    matchingSocket.on('partner-left', (data) => {
+      console.log('data received from socket', data)
+      if (data === 'partner left') {
+        navigate('/home')
+      }
+    })
+  }, [navigate])
```

Comment:
<img width="393" alt="image" src="https://user-images.githubusercontent.com/12640713/194686633-7e2e8657-dfc3-424f-8fec-fadce7d0960f.png">
Since I'm using navigate, I should include it as a dependency inside. Otherwise a warning will be thrown